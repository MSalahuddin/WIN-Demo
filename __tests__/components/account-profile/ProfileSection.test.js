import React from 'react';
import { View } from 'react-native';
import { render, fireEvent } from 'react-native-testing-library';

import ProfileSection from '../../../js/components/account-profile/ProfileSection';
import MockProvider from '../../MockProvider';

describe('Component ProfileSection', () => {
  const onPress = jest.fn();

  test('renders correctly', () => {
    const output = render(
      <MockProvider>
        <ProfileSection onPress={onPress} bannerTitle="" buttonTitle="">
          <View />
        </ProfileSection>
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('renders correctly when hideButton={true}', () => {
    const output = render(
      <MockProvider>
        <ProfileSection onPress={onPress} bannerTitle="" buttonTitle="" hideButton>
          <View />
        </ProfileSection>
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('renders correctly with banner values', () => {
    const output = render(
      <MockProvider>
        <ProfileSection
          onPress={onPress}
          bannerTitle=""
          buttonTitle=""
          bannerType="NORMAL"
          backdropSubtext=""
          backdropText=""
          hideButton
        >
          <View />
        </ProfileSection>
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('calls onPress when SectionButton is pressed', () => {
    const { getByTestId } = render(
      <MockProvider>
        <ProfileSection onPress={onPress} bannerTitle="" buttonTitle="">
          <View />
        </ProfileSection>
      </MockProvider>
    );
    fireEvent(getByTestId('section-button'), 'press');
    expect(onPress).toHaveBeenCalled();
  });
});
