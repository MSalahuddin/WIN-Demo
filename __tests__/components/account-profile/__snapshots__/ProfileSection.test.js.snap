// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ProfileSection renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-around",
        "marginTop": 40,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
    width={727.5}
  >
    <Image
      bannerType="BOW_TIE"
      resizeMode="contain"
      source={1}
      style={
        Array [
          Object {
            "height": 268.85869565217394,
            "width": 727.5,
          },
        ]
      }
      width={727.5}
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "space-between",
          "marginVertical": 48,
        },
      ]
    }
  >
    <View />
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginHorizontal": 10,
          "marginVertical": 10,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E5E5E5",
          "borderRadius": 80,
          "height": 110,
          "justifyContent": "center",
          "marginTop": 40,
          "opacity": 1,
          "width": 320,
        }
      }
      testID="section-button"
    >
      <View
        borderRadius={40}
        colorScheme="#FFFFFF"
        height={55}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 80,
              "height": 100,
              "justifyContent": "center",
              "width": 314,
            },
          ]
        }
        width={160}
      >
        <View
          borderRadius={40}
          colorScheme="#F9F9F9"
          height={55}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#F9F9F9",
                "borderRadius": 100,
                "height": 96,
                "justifyContent": "center",
                "marginTop": 8,
                "width": 314,
              },
            ]
          }
          width={160}
        >
          <Text
            alignCenter={false}
            allowFontScaling={true}
            color="#000000"
            fontFamily="MEDIUM"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="XSMALL"
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "Calibre-Medium",
                  "fontSize": 24,
                  "justifyContent": "center",
                  "textAlign": "left",
                  "textDecorationColor": "#000000",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Component ProfileSection renders correctly when hideButton={true} 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-around",
        "marginTop": 40,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
    width={727.5}
  >
    <Image
      bannerType="BOW_TIE"
      resizeMode="contain"
      source={1}
      style={
        Array [
          Object {
            "height": 268.85869565217394,
            "width": 727.5,
          },
        ]
      }
      width={727.5}
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "space-between",
          "marginVertical": 48,
        },
      ]
    }
  >
    <View />
  </View>
</View>
`;

exports[`Component ProfileSection renders correctly with banner values 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-around",
        "marginTop": 40,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
    width={727.5}
  >
    <Image
      bannerType="NORMAL"
      resizeMode="contain"
      source={1}
      style={
        Array [
          Object {
            "height": 139.17391304347825,
            "width": 727.5,
          },
        ]
      }
      width={727.5}
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "space-between",
          "marginVertical": 48,
        },
      ]
    }
  >
    <View />
  </View>
</View>
`;
